generators/aws/lib/aws.js

```diff
@@ -14,18 +13,7 @@ var AwsFactory = module.exports = function AwsFactory(generatorRef, cb) {
         cb();
     } catch (e) {
         generator.log('Installing AWS dependencies into your JHipster folder');
-        var jhipsterPath = 'node_modules/generator-jhipster';
-        var skipClient = generator.config.get('skipClient');
-        // use the global JHipster for apps with no client-side code
-        if (skipClient) {
-            var prefix = shelljs.exec('npm config get prefix', {silent:true}).trim();
-            if (os.platform() === 'win32') {
-                jhipsterPath = prefix + '/' + jhipsterPath;
-            } else {
-                jhipsterPath = prefix + '/lib/' + jhipsterPath;
-            }
-        }
-        shelljs.exec('npm install aws-sdk progress node-uuid --prefix ' + jhipsterPath, {silent:true}, function (code, msg, err) {
+        shelljs.exec('npm install aws-sdk progress uuid --prefix node_modules/generator-jhipster', {silent: true}, function (code, msg, err) {
```

Comment:
Indeed, there should be a check for this. And as Yarn is now our default, we should use Yarn by default, and switch to NPM if it is configured in the project